Chester/QueryBuilder.swift

```diff
@@ -70,7 +70,7 @@ public final class QueryBuilder {
   /// - Throws: `MissingCollection` if no collection is defined before passing in arguments
   public func with(arguments: Argument...) throws -> Self {
     guard let _ = queries.first else { throw QueryError.missingCollection }
-    self.queries[0].with(arguments: arguments)
+    self.queries[self.queries.indices.last!].with(arguments: arguments)
```

Comment:
with this change, the `guard` above serves no purpose, may I suggest that instead you unwrap the last index and that also saves us the force unwrap here:
```suggestion
    guard let lastIndex = queries.indices.last else {
      throw QueryError.missingCollection
    }
    queries[lastIndex].with(arguments: arguments)
```

Same for all the other cases.